pkg/query/querier.go

```diff
@@ -260,6 +265,18 @@ func (q *querier) selectFn(ctx context.Context, hints *storage.SelectHints, ms .
 		return nil, errors.Wrap(err, "convert matchers")
 	}
 
+	tenantSpecified := false
+	for _, sms := range sms {
+		if sms.Name == "tenant" {
```

Comment:
Here, we could immidately filter out if someone is asking for another tenant data?